src/main/java/seedu/address/model/social/UniqueSocialInfoList.java

```diff
@@ -0,0 +1,145 @@
+package seedu.address.model.social;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.commons.exceptions.DuplicateDataException;
+
+/**
+ * A list of social media information that enforces no nulls and
+ * no duplicate social info types between its elements.
+ */
+public class UniqueSocialInfoList implements Iterable<SocialInfo> {
+
+    private final ObservableList<SocialInfo> internalList = FXCollections.observableArrayList();
+
+    public UniqueSocialInfoList() {}
+
+    public UniqueSocialInfoList(Set<SocialInfo> socialInfos) {
+        requireAllNonNull(socialInfos);
+        internalList.addAll(socialInfos);
+
+        assert internalListSocialTypesUnique();
+    }
+
+
+    /**
+     * Returns all the social media information in this list as a Set.
+     */
+    public Set<SocialInfo> toSet() {
+        assert internalListSocialTypesUnique();
+        return new HashSet<>(internalList);
+    }
+
+    /**
+     * Replaces the SocialInfo in this list with those in the argument SocialInfo list.
+     * @param socialInfos
+     */
+    public void setSocialInfos(Set<SocialInfo> socialInfos) {
+        requireAllNonNull(socialInfos);
+        internalList.setAll(socialInfos);
+
+        assert internalListSocialTypesUnique();
+    }
+
+    /**
+     * Returns true if the list contains a SocialInfo with the same type
+     */
+    public boolean containsSameType(SocialInfo toCheck) {
+        requireNonNull(toCheck);
+
+        String toCheckType = toCheck.getSocialType();
+        for (SocialInfo socialInfo : internalList) {
+            String thisSocialType = socialInfo.getSocialType();
+            if (toCheckType.equals(thisSocialType)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Adds a SocialInfo to the list,
```

Comment:
Could add:
`@throws DuplicateSocialTypeException if the SocialInfo to add is a duplicate of an existing SocialInfo in the list.`